### ISSUE TITLE
XGboost regression model (and probabilistic forecasting faisability)

### ISSUE THREAD

**Issue by dumjax** [unit8co/darts#1134]
* Make sure to support probabilistic forecasts supported by xgboost
* it seems XGBoost also supports multi-outputs natively since version 1.6.2 - we should make sure we leverage this.

**Comment by yepher** [comment]
Do you have any idea when XGBoost will become available in Darts?

**Comment by hrzn** [comment]
No estimate yet, although it's fairly high on our backlog. Would you be interested in contributing this model @yepher ?

**Comment by yepher** [comment]
I am not familiar with the internals of Darts. But could be a fun project. I will take a look to see if it is something I could accomplish.

**Comment by yepher** [comment]
Here in [gradient_boosted_model.py#L7](https://github.com/unit8co/darts/blob/master/darts/models/forecasting/gradient_boosted_model.py#L7) the code says:

`...To enable LightGBM support in Darts, follow the detailed install instructions for LightGBM in the README...`

But when I look in the [README.m](https://github.com/unit8co/darts/blob/master/README.md) I do not see any mention of how to `install LightGGM`

**Comment by beckernick** [comment]
Would love to see XGBoost in darts! On the topic of multi-output regression, the [documentation](https://xgboost.readthedocs.io/en/stable/tutorials/multioutput.html) notes that multi-output has "limited support from objectives and metrics".

If there are certain objective functions that are important for darts but not currently available, we'd love to learn more. cc @trivialfs @dantegd

**Comment by trivialfis** [comment]
Hi, I help maintain the XGBoost project and developed the initial support for multi-output. Please let me know if I can be useful here. Would love the see XGBoost being part of darts.

**Comment by hrzn** [comment]
Hi @beckernick @trivialfis, I have started some work to include XGBoost in Darts. I haven't yet played with multi-output regression but this is not super duper important: we are wrapping several models in a MultiOutputRegressor by default so things will work as intended even on multivariate series or for multi-step forecasting.

One thing that I think would be very valuable, though, would be to have a way to do quantile regression, e.g., by having an `objective` allowing for quantile loss (sometimes called pinball loss). I haven't found anything like it in the documentation. Do you know if xgboost has a feature along those lines that I maybe missed?
Thanks!

**Comment by hrzn** [comment]
> Hi @beckernick @trivialfis, I have started some work to include XGBoost in Darts. I haven't yet played with multi-output regression but this is not super duper important: we are wrapping several models in a MultiOutputRegressor by default so things will work as intended even on multivariate series or for multi-step forecasting.
> 
> One thing that I think would be very valuable, though, would be to have a way to do quantile regression, e.g., by having an `objective` allowing for quantile loss (sometimes called pinball loss). I haven't found anything like it in the documentation. Do you know if xgboost has a feature along those lines that I maybe missed? Thanks!

I think a solution could be to specify a custom `objective` with a function implementing the quantile loss (such as [this one](https://gist.github.com/Nikolay-Lysenko/06769d701c1d9c9acb9a66f2f9d7a6c7#file-xgb_quantile_loss-py-L25)). It'd still be nicer if XGBoost supported quantile regression out of the box of course :)

**Comment by trivialfis** [comment]
We are planning for quantile regression support. Stay tuned. ;-)
